controllers/tcproute_controller.go

```diff
@@ -0,0 +1,231 @@
+	// TODO: enable deletion grace period https://github.com/Kong/blixt/issues/48
+	if tcproute.DeletionTimestamp != nil {
+		return ctrl.Result{}, r.ensureTCPRouteDeletedInDataPlane(ctx, tcproute, gateway)
+	}
+
+	// in all other cases ensure the TCPRoute is configured in the dataplane
+	if err := r.ensureTCPRouteConfiguredInDataPlane(ctx, tcproute, gateway); err != nil {
+		if err.Error() == "endpoints not ready" {
+			r.log.Info("endpoints not yet ready for TCPRoute, retrying", "namespace", tcproute.Namespace, "name", tcproute.Name)
+			return ctrl.Result{RequeueAfter: time.Second}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// isTCPRouteManaged verifies wether a provided TCPRoute is managed by this
+// controller, according to it's Gateway and GatewayClass.
+func (r *TCPRouteReconciler) isTCPRouteManaged(ctx context.Context, tcproute gatewayv1alpha2.TCPRoute) (bool, *gatewayv1beta1.Gateway, error) {
+	log := log.FromContext(ctx)
+
+	var supportedGateways []gatewayv1beta1.Gateway
+
+	//Use the retrieve objects its parent ref to look for the gateway.
+	var parentRefErr error
+	for _, parentRef := range tcproute.Spec.ParentRefs {
+
+		//Build Gateway object to retrieve
+		gw := new(gatewayv1beta1.Gateway)
+
+		ns := tcproute.Namespace
+		if parentRef.Namespace != nil {
+			ns = string(*parentRef.Namespace)
+		}
+
+		//Get Gateway for TCP Route
+		if err := r.Get(ctx, types.NamespacedName{Name: string(parentRef.Name), Namespace: ns}, gw); err != nil {
+			log.Info("Unable to get parent ref gateway for", "tcpRoute", parentRef.Name, "Namespace", ns, "err", err)
+			parentRefErr = err
+			//Check next parent ref.
+			continue
+		}
+
+		//Get GatewayClass for the Gateway and match to our name of controler
+		gwc := new(gatewayv1beta1.GatewayClass)
+		if err := r.Get(ctx, types.NamespacedName{Name: string(gw.Spec.GatewayClassName), Namespace: ns}, gwc); err != nil {
+			log.Info("Unable to get Gatewayclass", "Gateway", gw.Name, "Udp Route", parentRef.Name, "err", err)
+			parentRefErr = err
+			//Check next parent ref.
+			continue
+		}
+
+		if gwc.Spec.ControllerName != vars.GatewayClassControllerName {
+			//Check next parent ref.
+			continue
+		}
+
+		//Check if referred gateway has the at least one listener with properties defined from TCPRoute parentref.
+		if err := r.verifyListener(ctx, gw, parentRef); err != nil {
+			log.Info("No matching listener found for referred gateway", "GatewayName", parentRef.Name, "GatewayPort", parentRef.Port)
+			//Check next parent ref.
+			continue
+		}
+
+		supportedGateways = append(supportedGateways, *gw)
+	}
+
+	//only used for logging
+	if len(supportedGateways) < 1 {
+		return false, nil, parentRefErr
+	}
+
+	refferedGateway := &supportedGateways[0]
+	log.Info("TCP Route appeared referring to Gateway", "Gateway ", refferedGateway.Name, "GatewayClass Name", refferedGateway.Spec.GatewayClassName)
+
+	return true, refferedGateway, nil
```

Comment:
Why do we want to return only the first Gateway of the list? Shouldn't we take into account all the matching Gateways?

Reply:
Yes, but we're pacing ourselves to add support for multiple `Gateways` later as we don't need it right this moment for testing, and it provides an opportunity for a community member to jump in and work on something that's not huge. @rushi47 created https://github.com/Kong/blixt/issues/40 as a follow up for this, but it looks like we forgot to link that here. I've added the TODO with the link to both controllers in:

497953abbe09cd0052cf78086422b15092235bcc